numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
I don't understand. The purpose of this context manager is to 1) create a temp directory, 2) get a new temp filename in that directory, and 3) remove that directory when the context ends. What name are we supplying from where to where? How does this help?